tests/contrib/hooks/test_aws_athena_hook.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# 'License'); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import unittest
+from unittest import mock
+
+try:
+    from airflow.contrib.hooks.aws_athena_hook import AWSAthenaHook
+except ImportError:
+    AWSAthenaHook = None  # type: ignore
+
+
+class MockAthenaClient:
```

Comment:
I am not a fan of this, because you have basically two `AthenaHook`s you need to maintain.
If you use `mock.patch` you can achieve the same without having two classes to keep in sync.

Reply:
I agree. We should use `unittest.mock` in this place.

Reply:
I had an issue mocking this function because of the way boto3 implements the api. It's not a defined function but an api call. There are a few suggestions that I found [here:](https://stackoverflow.com/questions/37143597/mocking-boto3-s3-client-method-python)but we would still have to maintain it. The other aws hooks that use boto3 are able to use the[moto](https://github.com/spulec/moto) package.  but athena isn't mocked yet

Reply:
@mik-laj @feluelle  is there a solution for this ? Did some checks everyone everywhere complains it's impossible to mock athena with moto. How shall we proceeded?

Reply:
You don't need the moto package to mock a class. Just mock the calls that are accessing the boto library in the get_query_results function. So that it will never really call the content of the moto get_query_results function.

Reply:
Actually you only need to add `mock_conn.get_query_results.return_value = self.base_query_result` to the beginning of the test function. That is basically what you are already doing.

You are doing good actually. But I would remove the extra class.
This part
```
        if NextToken is None:
            return self.query_result_with_next
        else:
            return self.base_query_result
```
is logic that you don't need if you set the `mock_conn.get_query_results.return_value` directly in the test function.

Do you understand what I mean? :D

Reply:
Fixed!